### DISCUSSION TITLE
How to stop a workflow after a certain number of trials have completed?

### DISCUSSION THREAD

**Discussion by abeWustl** [bonsai-rx/bonsai#1427] [Q&A]
[2AA_6-20-23.zip](https://github.com/bonsai-rx/bonsai/files/11805781/2AA_6-20-23.zip)

I'm working on an active avoidance bonsai workflow in which the animal is place inside a shuttle box and it either has to avoid getting shocked during a tone or escape the shock a tone. One of the problems that I'm running into is that my workflow keeps starting a new trials and I don't know how to stop the delivery of new trials (e.g. stop delivering trials after 30 trials have completed).

In short, how can I stop the delivery of new trials after a set number of trials have been completed?

Attached is the workflow. It's messy so I apologize and please any feedback on how to clean up and streamline the workflow is appreciated!

Thanks!

P.S. inside the 2-AA node is the workflow
